Python/chapter02/2.8 - Loop Detection/miguel_soln_2.8.py

```diff
@@ -183,8 +183,20 @@ def loop_detection_linear_time_const_space(ll: LinkedList):
     while slow_ptr and fast_ptr and fast_ptr.next:
         slow_ptr = slow_ptr.next
         fast_ptr = fast_ptr.next.next
-        if slow_ptr is fast_ptr or slow_ptr is fast_ptr.next:
-            return slow_ptr
+        if slow_ptr is not fast_ptr:
```

Comment:
would it make sense to `break` upon reaching the opposite condition? or something like

```python
if fast_ptr is None:
    return None
if slow_ptr is fast_ptr:
    # if we get here, then there is a cycle.
    break
# ... the rest, to avoid nesting
```